### PR TITLE
recipe for windowing use cases

### PR DESCRIPTION
**Pull request by jancama2**
I think the desc is not necessary.

### CODE REVIEW

File: pages/implementation/LongDataLists.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Displaying long data lists
+
+Lets say you have a list of 500 items. You don't want them to be rendered all the time because it will impact the performance. There is a technique called _windowing_ that can be used to optimize the performance. In windowing, you render only items that can be currently seen on the screen.
```
Comment (horaklukas) [inline pages/implementation/LongDataLists.md]: Don't you have a link to page where this technique is described? Where did you get know about it? The link would be nice here..

File: pages/implementation/LongDataLists.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    )}
+  </AutoSizer>
+);
+```
+
+### Dynamic Data in `itemData` With Variable `itemSize`
+
+When you use `VariableSizeList`, you have to set `itemSize` prop to be a function that accepts an index and returns the size. `react-window` then creates styles for the items and caches the styles. The problem is that these styles are cached unless you invalidate them, does't matter if `itemData` change or not.
+
+To invalidate the styles when your `itemData` change you can use api of the list and react lifecycle methods.
```
Comment (horaklukas) [inline pages/implementation/LongDataLists.md]: lifecycle methods (or hooks as in the example)

File: pages/implementation/LongDataLists.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+We don't want to reinvent the wheel. We do want to use tested libraries that implement _windowing_. There are two well known libraries solving the same:
+
+- [react-virtualized](https://www.npmjs.com/package/react-virtualized) the former library, solves a lot of use cases
+- [react-window](https://www.npmjs.com/package/react-window) the latter library, created by the author of `react-virtualized`, lightweight solution supporting only horizontal/vertical lists and grids with fixed and variable item size.
+
+**The author of the libraries recommends using `react-window` everytime it solves your use cases.**
+
+## Use Cases
+
```
Comment (horaklukas) [inline pages/implementation/LongDataLists.md]: Here would be good some introduction, like. When you use `react-window`, you can choose from two components: `FixedSizeList` and `VaribleSizeList`

File: pages/implementation/LongDataLists.md (hunk 1 above)
Comment (jancama2) [reply pages/implementation/LongDataLists.md]: https://reactjs.org/docs/optimizing-performance.html#virtualize-long-lists

Comment (horaklukas) [reply pages/implementation/LongDataLists.md]: Great, so make word _windowing_ a link to this page.

Comment (jancama2) [reply pages/implementation/LongDataLists.md]: Done

**Pull request merged**